log/logger.go

```diff
@@ -30,15 +27,59 @@ func init() {
 }
 
 // ModuleKey defines a module logging key.
-const ModuleKey = corelog.ModuleKey
+const ModuleKey = "module"
 
 // ContextKey is used to store the logger in the context.
 var ContextKey struct{}
 
 // Logger is the Cosmos SDK logger interface.
 // It maintains as much backward compatibility with the CometBFT logger as possible.
 // All functionalities of the logger are available through the Impl() method.
-type Logger = corelog.Logger
+type Logger interface {
+	LoggerBase
+
+	// With returns a new wrapped logger with additional context provided by a set.
+	With(keyVals ...any) Logger
+}
+
+// LoggerV2 is the new Cosmos SDK logger interface. It can be used in libraries without any need to import
+// the core or log packages directly.
+// All constructors in this package return a logger that implements this interface, but the constructor
+// functions have not been changed to avoid breakage.
+type LoggerV2 interface {
+	LoggerBase
+
+	// WithContext returns a new wrapped logger with additional context provided by the key value pairs.
+	// The returned value can be safely cast to LoggerV2. An any is returned instead of LoggerV2
+	// to avoid the need for log users to import the log package directly.
+	WithContext(keyVals ...any) any
+}
```

Comment:
can we make these internal? it will be confusing to see which one to use

Reply:
Meaning put them in an internal package? We can just avoid declaring the interface at all and then do it in log/v2 if we want